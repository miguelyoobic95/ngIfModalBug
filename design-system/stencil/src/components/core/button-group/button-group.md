---
name: Button Group
category: Core Components
---

`<yoo-button>`'s groups can be organized with `<yoo-button-group>` HTML tag.

## Attributes

|Name|Type|Default|Description|
|---|---|---|----|
|isDropdown|boolean|`false`|True creates a dropdown button group|
|dropdownTitle|string||Specifies the text to display in the dropdown button|

```yoo-button-group.html
    <yoo-button-group is-dropdown="true" dropdown-title="Button Dropdown">
        <yoo-button text="button 1" class="accent"></yoo-button>
        <yoo-button text="button 2 long text" class="dark"></yoo-button>
        <yoo-button text="button 3"></yoo-button>
        <yoo-button class="icon-only"  class="accent" icon="yo-fire"></yoo-button>
        <yoo-button text="button 4" class="dark"></yoo-button> 
    </yoo-button-group>
```

### CSS

|Type|Name|Description|
|---|---|---|
|Horizontal| default| The default groups buttons horizontally|
|Vertical| `vertical`| Group buttons vertically|
|Justified|`justified`| Evenly separates the buttons in the group

```yoo-button-group-class.html
    <yoo-button-group>
        <yoo-button text="button 1" class="accent"></yoo-button>
        <yoo-button text="button 2 long text" class="dark"></yoo-button>
        <yoo-button text="button 3"></yoo-button>
        <yoo-button class="icon-only"  class="accent" icon="yo-fire"></yoo-button>
        <yoo-button text="button 4" class="dark"></yoo-button> 
    </yoo-button-group>
    <br/>
     <yoo-button-group class="vertical">
        <yoo-button text="button 1"></yoo-button>
        <yoo-button text="button 2"></yoo-button>
        <yoo-button text="fire" class="accent" icon="yo-fire"></yoo-button>
    </yoo-button-group>
    <br/>
    <yoo-button-group class="justified">
        <yoo-button text="button 1"></yoo-button>
        <yoo-button text="button 2 long text" class="secondary"></yoo-button>
        <yoo-button text="button 4"></yoo-button>
    </yoo-button-group>
```