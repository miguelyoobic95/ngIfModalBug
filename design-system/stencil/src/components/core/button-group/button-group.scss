

$dark-border-color: cssvar(dark-40);
:host {
    // Horizontal Button Group
    .group-container {
        /deep/:nth-child(n) {
            .container {
                border-radius: 0px;
                @include color-border(right, 1px, $dark-border-color);
                @include color-border(left, 0px, $dark-border-color);
            }
        }
        /deep/:first-child {
            .container {
                @include border-left-radius(5px);
                @include color-border(left, 1px, $dark-border-color);
                @include color-border(right, 1px, $dark-border-color);
            }
        }
        /deep/:last-child {
            .container {
                @include border-right-radius(5px);
                @include color-border(right, 1px, $dark-border-color);
                @include color-border(left, 0px, $dark-border-color);
            }
        }
        /deep/.container {
            @include color-border(top, 1px, $dark-border-color);
            @include color-border(bottom, 1px, $dark-border-color);
            display: inline-flex;
            justify-content: center;
            &:hover {
                box-shadow: none;
                transform: none;
            }
        }
    }
    yoo-context-menu {
        /deep/yoo-button {
            span {
                float: right;
                padding-left: 0.5rem;
                padding-right: 0.5rem;
            }
        }
    }
    .context-container {
        padding-left: 10px;
        padding-top: 5px;
        padding-bottom: 5px;

        yoo-button {
            width: inherit;
        }
    }
}

:host(.vertical) {
    .group-container {
        display: flex;
        flex-direction: column;
        width: 100%;
        /deep/:nth-child(n) {
            .container {
                border-radius: 0px;
                @include color-border(bottom, 1px, $dark-border-color);
            }
        }
        /deep/:first-child {
            .container {
                @include color-border(top, 1px, $dark-border-color);
                @include color-border(bottom, 1px, $dark-border-color);
                @include border-top-radius(5px);
            }
        }
        /deep/:last-child {
            .container {
                @include color-border(bottom, 1px, $dark-border-color);
                @include border-bottom-radius(5px);
            }
        }
        /deep/.container {
            @include color-border(right, 1px, $dark-border-color);
            @include color-border(left, 1px, $dark-border-color);
            border-top: none;
            border-bottom: none;
        }
    }
}

:host(.justified) {
    .group-container {
        justify-content: space-between;
        /deep/.container {
            border: 1px solid cssvar(dark-40);
            border-radius: 5px;
        }
    }
}