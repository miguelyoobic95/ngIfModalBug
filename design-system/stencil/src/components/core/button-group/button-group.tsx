import { Component, Prop, Element } from '@stencil/core';

@Component({
    tag: 'yoo-button-group',
    styleUrl: 'button-group.scss',
    scoped: true
})
export class YooButtonGroupComponent {

    @Prop() isDropdown: boolean = false;
    @Prop() dropdownTitle: string;

    @Element() host: HTMLStencilElement;

    render(): JSX.Element {
        return (
            (this.isDropdown ? <yoo-context-menu>
                <yoo-button slot="trigger" icon="yo-arrow-dropdown" text={this.dropdownTitle}></yoo-button>
                <div class="context-container" attr-layout="column">
                    <slot/>
                </div>
            </yoo-context-menu>
            : <div class="group-container" attr-layout="row">
                <slot/>
            </div>)
        );
    }
}
