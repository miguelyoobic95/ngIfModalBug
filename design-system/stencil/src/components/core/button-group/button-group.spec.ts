import { TestWindow } from '@stencil/core/testing';
import { YooButtonGroupComponent } from './button-group';

describe('YooButtonGroupComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooButtonGroupComponent()).toBeTruthy();
    });
    describe('Snapshots', () => {
        it('Should project content correctly', async () => {
            let element = await window.load({
                components: [YooButtonGroupComponent],
                html: `<yoo-button-group>
                <yoo-button text="button 1"></yoo-button>
                <yoo-button text="button 2"></yoo-button>
                </yoo-button-group>`
            });
            expect(element).toMatchSnapshot();
        });
        it('Should create a dropdown button with the correct title', async () => {
            let element = await window.load({
                components: [YooButtonGroupComponent],
                html: `<yoo-button-group is-dropdown="true" dropdown-title="Button dropdown>
                <yoo-button text="button 1"></yoo-button>
                <yoo-button text="button 2"></yoo-button>
                </yoo-button-group>`
            });
            expect(element).toMatchSnapshot();
        });
    });
    describe('Rendering', () => {
        it('Should render with 3 buttons', async () => {
            let element = await window.load({
                components: [YooButtonGroupComponent],
                html:  `<yoo-button-group>
                            <yoo-button text="button 1"></yoo-button>
                            <yoo-button text="button 2"></yoo-button>
                            <yoo-button text="button 3"></yoo-button>
                        </yoo-button-group>`
             });

             expect(element.getElementsByTagName('yoo-button').length).toBe(3);
        });
    });
});
