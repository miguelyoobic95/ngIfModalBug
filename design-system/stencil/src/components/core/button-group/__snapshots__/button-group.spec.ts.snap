// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooButtonGroupComponent Snapshots Should create a dropdown button with the correct title 1`] = `
<yoo-button-group
  1"=""
  button=""
  class="hydrated"
  data-yoo-button-group-host=""
  dropdown-title="Button dropdown>
                <yoo-button text="
  is-dropdown="true"
>
  
  
  
  
  <yoo-context-menu
    data-yoo-button-group=""
  >
    <yoo-button
      data-yoo-button-group=""
      icon="yo-arrow-dropdown"
      slot="trigger"
      text="Button dropdown>
                <yoo-button text="
    />
    <div
      attr-layout="column"
      class="context-container"
      data-yoo-button-group=""
      data-yoo-button-group-slot=""
    >
      
      
                
      <yoo-button
        text="button 2"
      />
      
                
    </div>
  </yoo-context-menu>
</yoo-button-group>
`;

exports[`YooButtonGroupComponent Snapshots Should project content correctly 1`] = `
<yoo-button-group
  class="hydrated"
  data-yoo-button-group-host=""
>
  
  
  
  
  
  
  <div
    attr-layout="row"
    class="group-container"
    data-yoo-button-group=""
    data-yoo-button-group-slot=""
  >
    
    
                
    <yoo-button
      text="button 1"
    />
    
                
    <yoo-button
      text="button 2"
    />
    
                
  </div>
</yoo-button-group>
`;
